garnett/translatedstr.py

```diff
@@ -0,0 +1,82 @@
+from typing import Tuple, Optional, Callable
+from django.utils.translation import gettext as _
+
+from langcodes import Language
+from garnett.utils import get_current_language, get_current_language_code, get_languages
+
+
+class HTMLTranslationMixin:
+    def __html__(self) -> str:
+        # Add leading [lang] wrapped in a span
+        text = self
+        if not self.is_fallback:
+            return text
+        elif not self.fallback_language:
+            return "??"
+        else:
+            return (
+                '<span class="fallback"'
+                'data-lang-code="{lang.language}"'
+                ">"
+                "[{lang.language}]</span> "
+                "{s}"
+            ).format(s=self, lang=self.fallback_language)
+
+
+class TranslatedStr(str):
+    def __new__(cls, content, fallback: Callable = None):
+        current_language_code = get_current_language_code()
+        has_current_language = current_language_code in content.keys()
+        if has_current_language:
+            is_fallback = False
+            fallback_language = None
+            text = content.get(current_language_code)
+        else:
+            is_fallback = True
+            if fallback:
+                fallback_language, text = fallback(content)
+            else:
+                fallback_language, text = cls.get_fallback_text(content)
+
+        instance = super().__new__(cls, text)
+        instance.translations = content
+        instance.is_fallback = is_fallback

```

Comment:
You don't need to assign is_fallback in your check - it would be cleaner to assign it directly to `instance.is_fallback`. 

And you could shorten it by setting fallback_language to None outside of the second if statement - which is then reassigned if not has_current_language